include/maths/CKdTree.h

```diff
@@ -512,6 +550,11 @@ class CKdTree {
         return coordinate == m_Dimension ? 0 : coordinate;
     }
 
+    //! Cast away any reference wrapper.
+    inline static const TPoint& unwrap(const POINT& p) {
+        return static_cast<const TPoint&>(p);
+    }
```

Comment:
Could `core::unwrap_ref()` from `core/UnwrapRef.h` do this?

Reply:
See [dbb74db](https://github.com/elastic/ml-cpp/pull/1070/commits/dbb74dbcd91239394e2ae1bada60489a4cd00cef).